lib/internal/repl/await.js

```diff
@@ -92,13 +97,38 @@ for (const nodeType of ObjectKeys(walk.base)) {
 }
 
 function processTopLevelAwait(src) {
-  const wrapped = `(async () => { ${src} })()`;
+  const wrapPrefix = '(async () => {\n';
+  const wrapped = `${wrapPrefix}${src}\n})()`;
   const wrappedArray = ArrayFrom(wrapped);
   let root;
   try {
     root = parser.parse(wrapped, { ecmaVersion: 'latest' });
-  } catch {
-    return null;
+  } catch (e) {
+    // If the parse error is before the first "await", then use the execution
+    // error. Otherwise we must emit this parse error, making it look like a
+    // proper syntax error.
+    const awaitPos = src.indexOf('await');
+    const errPos = e.pos - wrapPrefix.length;
+    if (awaitPos > errPos)
+      return null;
+    // Convert keyword parse errors on await into their original errors when
+    // possible.
+    if (errPos === awaitPos + 6 &&
+        StringPrototypeIncludes(e.message, 'Expecting Unicode escape sequence'))
+      return null;
+    if (errPos === awaitPos + 7 &&
+        StringPrototypeIncludes(e.message, 'Unexpected token'))
+      return null;
+    const { line, column } = e.loc;
+    let message = '\n' + StringPrototypeSplit(src, '\n')[line - 2] + '\n';
+    let i = 0;
+    while (i++ < column) message += ' ';
```

Comment:
Why use a loop here?
```suggestion
    let message = '\n' + StringPrototypeSplit(src, '\n')[line - 2] + '\n' + StringPrototypeRepeat(' ', column);
```